synchronization/src/main/java/de/cyface/synchronization/SyncAdapter.kt

```diff
@@ -124,189 +128,191 @@ class SyncAdapter private constructor(
-                                "Measurement marked as %s.",
-                                result.toString().lowercase()
+                                "Measurement with identifier %d is about to be loaded for transmission.",
+                                measurement!!.id
                             )
                         )
-                    } catch (e: NoSuchMeasurementException) {
-                        throw IllegalStateException(e)
+
+                        // Ensure the measurement is supported
+                        val format = measurement.fileFormatVersion
+                        Validate.isTrue(format == DefaultPersistenceLayer.PERSISTENCE_FILE_FORMAT_VERSION)
+
+                        // Load measurement data
+                        val metaData = loadMetaData(measurement, persistence, deviceId, context)
+
+                        // Load, try to sync the file to be transferred and clean it up afterwards
+                        var compressedTransferTempFile: File? = null
+                        try {
+                            runBlocking {
+                                compressedTransferTempFile =
+                                    serializer.writeSerializedCompressed(
+                                        measurement.id,
+                                        persistence
+                                    )
+                            }
+
+                            // Check whether the network settings changed to avoid using metered network without permission
+                            if (isSyncRequestAborted(account, authority) || error) {
+                                return@performActionWithFreshTokens
+                            }
+
+                            // Synchronize measurement
+                            val processListener = object : UploadProgressListener {
+                                override fun updatedProgress(percent: Float) {
+                                    // Multi-measurement progress
+                                    val progressPerMeasurement =
+                                        100.0 / measurementCount.toDouble()
+                                    val progressBeforeThis =
+                                        index.toDouble() * progressPerMeasurement
+                                    val lastMeasurement = index == measurementCount - 1
+                                    val total =
+                                        if (lastMeasurement && percent.toDouble() == 1.0) 100.0 else progressBeforeThis + percent * progressPerMeasurement
+                                    for (listener in progressListener) {
+                                        listener.onProgress(total.toFloat(), measurement.id)
+                                    }
+                                }
+                            }
+
+                            // This is now executed asynchronously!
```

Comment:
Make larger measurements with different devices and see if this is an issue (OOM or so)